### ISSUE TITLE
[ENH] Add `INCOMPLETE` and `UNAVAILABLE` as processing statuses

### ISSUE THREAD

**Issue by michellewang** [nipoppy/nipoppy#421]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### New feature

- `UNAVAILABLE` determined from the manifest
- `FAIL` based on latest log file
-  `INCOMPLETE` if not all expected files are there

**Comment by michellewang** [comment]
- Related to #460 for `UNAVAILABLE` status

**Comment by michellewang** [comment]
@julia-pfarr this is the issue I referred to previously. We might want to split this into sub-issues, and we should probably discuss this more deeply anyway. Currently the tracking logic is:
- All files in tracker config present --> `SUCCESS`
- Otherwise --> `FAIL`

Ideally we will have `FAIL` only if the pipeline crashed, but this is not the easiest thing to determine (will need to check log files, and even that might not be very robust). So I think we leave this for later.

For now the easiest thing would be to add an `INCOMPLETE` status. So we would have
- All files in tracker config present --> `SUCCESS`
- Otherwise:
   - Parent directory (either `workflow.dpath_pipeline_output` or `PARTICIPANT_SESSION_DIR` in the tracker config) does not exist --> `INCOMPLETE`
   - Otherwise --> `FAIL`

Let me know if you are interested in implementing that :)

**Comment by julia-pfarr** [comment]
Yes, I am interested in implementing thais Let's talk about it in more detail soon. I was also thinking a bit more about this status

> Parent directory (either `workflow.dpath_pipeline_output` or `PARTICIPANT_SESSION_DIR` in the tracker config) does not exist --> `INCOMPLETE`

because I'm a bit worried that `INCOMPLETE` is not less ambiguous than `FAIL` for this status. My suggestions are:

`UNEXECUTED`
`INITIALIZED`
`CONFIGURED`
`READY`
`AWAITING INITIATION`
`SETUP COMPLETE`

but for those we would also need to check if pipeline is even installed. 

so we could have:

- pipeline not installed: `INCOMPLETE/MISSING SETUP` or `UNAVAILABLE`
- pipeline installed but no  `workflow.dpath_pipeline_output` or `PARTICIPANT_SESSION_DIR` in the tracker config: `SETUP COMPLETE`
- pipeline run successfully: `SUCCESS`
- otherwise:` FAIL`

Re 

> UNAVAILABLE determined from the manifest

Do you mean with this that the pipeline cannot be run anyway because the datatype does not fit or do you mean with this that the pipeline is not installed?

**Comment by michellewang** [comment]
@julia-pfarr interesting! It makes sense to have more granularity in the processing status, happy to discuss more soon. Not entirely sure it makes sense to check if a pipeline is installed or not because if you can run the tracker then it is already installed.

>> UNAVAILABLE determined from the manifest
>
>Do you mean with this that the pipeline cannot be run anyway because the datatype does not fit or do you mean with this that the pipeline is not installed?

Cannot be run because the datatype does not fit (for example if not all participants have DWI data then we expect some of them to never have diffusion derivatives)

**Comment by julia-pfarr** [comment]
> Not entirely sure it makes sense to check if a pipeline is installed or not because if you can run the tracker then it is already installed.

Ah, I see! I didn't know, I've never tried this before. I thought it would also write a `FAIL` for this then. But I just tried and it says "no config file found for pipeline...", so that's already perfect.

**Comment by michellewang** [comment]
Was talking with @mathdugre about saving the Boutiques provenance file (stored in `~/.cache/boutiques/data`) in the pipeline output folder for each run. If we do that then we could get the exit status from that file too and we can make `FAILED` mean "non-zero exit code" that way

**Comment by yarikoptic** [comment]
claude says that `INCOMPLETE` and `UNAVAILABLE` are now valid statuses in `ProcessingStatusTable` (see `nipoppy/tabular/processing_status.py`), may be that is true and this could be closed?
